mantis-runtime/src/main/java/io/mantisrx/runtime/source/http/impl/HttpClientFactories.java

```diff
@@ -25,67 +25,108 @@
 import mantis.io.reactivex.netty.protocol.http.client.HttpClientBuilder;
 import mantis.io.reactivex.netty.protocol.http.sse.ServerSentEvent;
 
-
 public class HttpClientFactories {
 
     public static HttpClientFactory<ByteBuf, ByteBuf> defaultFactory() {
         return new DefaultHttpClientFactory();
     }
 
+    public static HttpClientFactory<ByteBuf, ByteBuf> defaultFactory(
+        boolean enableConnectionPooling,
+        boolean enableIdleConnectionCleanup) {
+        return new DefaultHttpClientFactory(enableConnectionPooling, enableIdleConnectionCleanup);
+    }
+
     public static HttpClientFactory<ByteBuf, ServerSentEvent> sseClientFactory() {
         return new SSEClientFactory();
     }
 
-    public static HttpClientFactory<ByteBuf, ServerSentEvent> sseClientFactory(int readTimeout) {
-        return new SSEClientFactory(readTimeout);
+    public static HttpClientFactory<ByteBuf, ServerSentEvent> sseClientFactory(
+        boolean enableConnectionPooling,
+        boolean enableIdleConnectionCleanup) {
+        return new SSEClientFactory(enableConnectionPooling, enableIdleConnectionCleanup);
     }
 
-
     private static class DefaultHttpClientFactory implements HttpClientFactory<ByteBuf, ByteBuf> {
 
         private final HttpClient.ClientConfig clientConfig;
+        private final boolean enableConnectionPooling;
```

Comment:
do you know what impact enabling/disabling these options will have on the jobs? Any particular side effects we should be aware about?

Reply:
In general connection pooling helps in certain cases for perf/resource control but in the actual use cases like the one in the source job it's essentially just connecting to a given ip+port per client instance thus the benefit seems insignificant.